File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    private int lr(String s){
        if(s.equals("left")) return -1;
        else if(s.equals("right")) return 1;
        return 0;
    }
    private int ud(String s){
        if(s.equals("down")) return -1;
        else if(s.equals("up")) return 1;
        return 0;
    }
    public int[] solution(String[] keyinput, int[] board) {
        int n = board[0]/2 == 0 ? 1 : board[0]/2;
        int m = board[1]/2 == 0 ? 1 : board[1]/2;
        int r = 0;
        int u = 0;
        for(String s : keyinput){
            int nr = r + lr(s); 
            int nu = u + ud(s);
            if(Math.abs(nr) > n || Math.abs(nu) > m) continue;
            r = nr;
            u = nu;
        }
        return new int[]{r, u};
    }
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    col, row = board
    col //= 2
    row //= 2
    x,y = 0, 0
    for cmd in keyinput:
        nx, ny = x, y
        if cmd == "left":
            nx -= 1
        elif cmd == "right":
            nx += 1
        elif cmd == "up":
            ny += 1
        elif cmd == "down":
            ny -= 1
        if abs(ny) > row or abs(nx) > col:
            continue
        x, y = nx, ny
    return [x, y]
# (0, 0) 시작, (x,y) 리턴 